tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py

```diff
@@ -40,25 +40,28 @@ def __init__(self, **kwargs):
     def _set_default_params(self):
         import torch
 
-        default_params = get_default_param(self.problem_type)
-        for param, val in default_params.items():
-            self._set_default_param_value(param, val)
+        default_params = dict()
 
         # TODO: Take in num_gpu's as a param. Currently this is hard-coded upon detection of cuda.
         if torch.cuda.is_available():
-            device = torch.device("cuda")
+            default_params['device'] = torch.device("cuda")
         else:
-            device = torch.device("cpu")
+            default_params['device'] = torch.device("cpu")
 
-        self.params['device'] = device
+        default_params.update(get_default_param(self.problem_type))
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
 
-    def get_model(self):
+    def _get_model(self):
         from .tab_model_base import TabNet
         # If we have already initialized the model, we don't need to do it again.
+        model = None
         if self.model is None:
```

Comment:
this check should probably be removed now that it is returning a model.

Reply:
Whoops. Got it!